src/mongocxx/client_encryption.cpp

```diff
@@ -50,6 +51,55 @@ bsoncxx::types::bson_value::value client_encryption::decrypt(
     return _impl->decrypt(value);
 }
 
+collection client_encryption::create_encrypted_collection(const database& db,
+                                                          const std::string& coll_name,
+                                                          const bsoncxx::document::view& options,
+                                                          bsoncxx::document::value& out_options,
+                                                          const std::string& kms_provider) {
+    return this->create_encrypted_collection(
+        db, coll_name, options, out_options, kms_provider, stdx::nullopt);
+}
+
+collection client_encryption::create_encrypted_collection(
+    const database& db,
+    const std::string& coll_name,
+    const bsoncxx::document::view& options,
+    bsoncxx::document::value& out_options,
+    const std::string& kms_provider,
+    const stdx::optional<bsoncxx::document::view>& masterkey) {
+    std::error_code ec;
+    auto coll = this->create_encrypted_collection(
+        db, coll_name, options, out_options, kms_provider, masterkey, ec);
+    if (!coll) {
+        throw operation_exception(ec, ec.message());
+    }
+    return *coll;
+}
+
+stdx::optional<collection> client_encryption::create_encrypted_collection(
+    const database& db,
+    const std::string& coll_name,
+    const bsoncxx::document::view& options,
+    bsoncxx::document::value& out_options,
+    const std::string& kms_provider,
+    std::error_code& ec) noexcept {
```

Comment:
Partially, especially because error handling is important with this API since it can create data keys and *then* fail, meaning that an unhandled error will generate unused data keys.

Reply:
Changed it